src/libawkward/io/json.cpp

```diff
@@ -303,16 +345,27 @@ namespace awkward {
 
   ToJsonFile::ToJsonFile(FILE* destination,
                          int64_t maxdecimals,
-                         int64_t buffersize)
-      : impl_(new ToJsonFile::Impl(destination, maxdecimals, buffersize)) { }
+                         int64_t buffersize,
+                         const char* nan_string,
+                         const char* infinity_string,
+                         const char* minus_infinity_string)
+      : impl_(new ToJsonFile::Impl(destination, maxdecimals, buffersize))
+      , nan_string_(nan_string)
+      , infinity_string_(infinity_string)
+      , minus_infinity_string_(minus_infinity_string) { }
 
   ToJsonFile::~ToJsonFile() {
     delete impl_;
   }
 
   void
   ToJsonFile::null() {
-    impl_->null();
+    if (nan_string_ != nullptr) {
+      impl_->string(nan_string_, (int64_t)strlen(nan_string_));
+    }
+    else {
+      impl_->null();
+    }
```

Comment:
For the same reason as above, please revert this.

Reply:
done